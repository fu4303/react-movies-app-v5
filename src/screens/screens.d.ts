interface MovieProps {
  popularity: number;
  vote_count: number;
  video: boolean;
  poster_path: string;
  id: number;
  adult: boolean;
  backdrop_path: string;
  original_language: string;
  original_title: string;
  genre_ids: number[];
  title: string;
  vote_average: number;
  overview: string;
  release_date: string;
  isFavorite?: boolean;
  isOnWatchList?: boolean;
}

interface SearchMoviesProps {
  page: number;
  total_results: number;
  total_pages: number;
  results: MovieProps[];
}

interface MovieDetails extends MovieProps {
  isOnWatchList?: boolean;
  isFavorite?: boolean;
  imdb_id: string;
  tagline: string;
  videos: {
    results: MovieVideoProps[]
  }
  homepage: string;
  genres: MovieGenre[];
}

interface MovieVideoProps {
  id: string,
  iso_639_1: string,
  iso_3166_1: string,
  key: string,
  name: string,
  site: string,
  size: number,
  type: string
}

interface MovieGenre {
  id: number;
  name: string;
}
