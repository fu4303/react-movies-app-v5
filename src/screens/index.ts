export * from './HomePage/HomePage';
export * from './SearchPage/SearchPage';
export * from './MovieDetailsPage/MovieDetails';
