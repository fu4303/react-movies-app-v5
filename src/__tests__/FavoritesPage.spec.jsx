import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { FavoritesPage } from 'screens';

afterEach(cleanup);

describe('Favorites Page tests', () => {
  it('Should match the snapshot for "FavoritesPage" screen', () => {
    const { asFragment } = render(<FavoritesPage />);
    expect(asFragment(<FavoritesPage />)).toMatchSnapshot();
  });
});
