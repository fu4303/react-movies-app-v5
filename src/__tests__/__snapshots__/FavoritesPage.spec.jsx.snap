// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites Page tests Should match the snapshot for "FavoritesPage" screen 1`] = `
<DocumentFragment>
  <h2
    class="sc-fzqBkg frbgpf"
  >
    My Favorite Movies
  </h2>
  <section
    class="sc-fznMAR kJIjAt"
  >
    <p
      class="sc-fzolEj bQKqPZ"
    >
      You favorite movies will appear here!
    </p>
    <img
      alt="You favorite movies will appear here!"
      class="sc-fznWOq kwxMvj"
      src="cinema.svg"
    />
  </section>
</DocumentFragment>
`;
